### DISCUSSION TITLE
Performance Comparison Using Interactive transactions VS multiple prisma.$transaction

### DISCUSSION THREAD

**Discussion by enosh15dev** [prisma/prisma#17373] [Q&A]
# In short: Which is better in performance between 

1. run `await prisma.$transaction(items)` multiple times

2. run `await prisma.$transaction(async (tx)=>{ FOR LOOP HERE }) `

===

>  https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide

While following the guide above, i got a curiosity.
i need to update 10 different items' count.
First i check if i have enough.
## Old Code
```javascript
const find_items = item_codes.map((x) =>
        prisma.item.findUnique({
          where: { code: x },
        }),
      )

await prisma.$transaction(find_items)
```
And if i have enough, i update.
```javascript
const update_items = arr.map((x) =>
        prisma.item.update({
          where: { code: x.code },
          data: {
            count: { decrement: x.count] },
          },
        }),
      )
await prisma.$transaction(update_items)
```

## New Code after prisma 4.5.0
 i found **Interactive transactions** and i think it's very cool.
but since i need to use `for loop` inside, i'm not sure if the new code works faster and safer.

```javascript 
await prisma.$transaction(async (tx) => {
      for (let i = 0; i < arr.length; i++) {
        const item = await tx.item.update({
          where: { ... },
          data: { ... },
        })
        if (item.count < 0) {
          throw new Error()
        }
      }
    })
```

Could someone answer me?

**Comment by nurul3101** [comment]
Hey @enosh15dev 👋 

Prisma supports multiple ways of handling transactions, either directly through the API or by supporting your ability to introduce optimistic concurrency control through interactive transactions.

For comparing performance, the best way to find out would be to add this [logging middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware/logging-middleware) and compare the time taken in both cases.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
